src/ui/components/LedgerAdmin.js

```diff
@@ -74,38 +87,60 @@ export const LedgerAdmin = (): ReactNode => {
     let nextLedger;
     if (ledger.account(id).active) {
       nextLedger = ledger.deactivate(id);
-      setCheckBoxSelected(false);
     } else {
       nextLedger = ledger.activate(id);
-      setCheckBoxSelected(true);
     }
     updateLedger(nextLedger);
   };
 
   const resetIdentity = () => {
     setIdentityName("");
     setSelectedId(null);
-    setCheckBoxSelected(false);
     setPromptString("Add Identity: ");
   };
 
   const setActiveIdentity = (identity: Identity) => {
     setIdentityName(identity.name);
     setSelectedId(identity.id);
-    setCheckBoxSelected(ledger.account(identity.id).active);
     setPromptString("Update Identity: ");
   };
 
+  const loadEntitiesCheckList = () => {
+    let identities = {};
+
+    ledger
+      .accounts()
+      .map((a) => a.identity)
+      .map((identity) => {
+        identities = {
+          ...identities,
+          [`${identity.id}`]: ledger.account(identity.id).active,
+        };
+      });
+
+    setIsChecked(identities);
+  };
+
+  useEffect(() => loadEntitiesCheckList(), [ledger]);
+
   const renderIdentities = () => {
     const renderIdentity = (i: Identity, notLastElement: boolean) => (
       <>
         <ListItem button onClick={() => setActiveIdentity(i)} key={i.id}>
           {i.name}
+          <Checkbox
+            onChange={handleSingleCheck}
+            checked={NullUtil.orElse(isChecked[i.id], false)}
```

Comment:
There may be other reasons that `NullUtil.orElse()` is used commonly in the codebase (I'm still learning a lot about how things are done here) but you should also be able to use:
```suggestion
            checked={Boolean(isChecked[i.id])}
```
Same goes for line 179 below. If so, you can scratch the import on line 21.